src/gccrs/args.rs

```diff
@@ -106,6 +115,54 @@ impl GccrsArgs {
         options
     }
 
+    fn generate_static_lib(args: &GccrsArgs) -> Result {
+        let output_file = args
+            .output_file
+            .to_str()
+            .expect("Cannot handle non-unicode filenames yet");
+
+        Command::new("ar")
+            .args(&[
+                "rcs", // Create the archive and add the files to it
+                output_file,
+                object_file_name(&output_file).as_str(),
+            ])
+            .status()?;
+
+        Ok(())
+    }
+
+    fn with_callback(self, function: &'static dyn Fn(&GccrsArgs) -> Result) -> GccrsArgs {
+        GccrsArgs {
```

Comment:
NIT: You can use `Self` instead of `GccrsArgs`

```suggestion
    fn with_callback(self, function: &'static dyn Fn(&Self) -> Result) -> Self {
        Self {
```

There's a Clippy lint for it, which you'd have to enable with `#![warn(clippy::use_self)]` in `lib.rs`.

This is a style question though, so feel free to just dismiss this comment.

Reply:
I've always used the type's name instead of `Self`, despite knowing about it. I think it looks a little bit clearer but if you have any reasoning as to why using `Self` would be better, I'm all ears!

Reply:
Personal preference. But as long as it's consistent, I'm fine with both.